File: exercicios em C/q29.cpp
#include <stdio.h>

int main(void){
	int num, i, resultado;
	resultado = 0;
 	printf("\n\nEste programa determina se um numero e primo ou nao.");
 	printf("\n\nDigite o numero desejado: ");
	scanf("%d", &num);
	for (i = 2; i <= num / 2; i++) {
		if (num % i == 0) {
	       resultado++;
	       break;
	    }
	}
	//se o senhor ta lendo isso quero deseja para o senhor um bom dia:) pq eu n tive na hora de fazer essa questao
	if (resultado == 0){
	    printf("\n\n%d e um numero primo\n", num);
	}
	else{
		printf("%d nao e um numero primo\n", num);
	}
	printf("\n\nObrigado por utilizar nosso programa!");
	return 0;
} 

File: exercicios em Python/q29.py
import os

print("\n\nEste programa determina se um numero e primo ou nao.")
num = int(input("\nDigite o numero desejado: "))
i = 2
resultado = 0
while i <= (num / 2) :
        if num % i == 0:
                resultado = resultado + 1
                break
        i = i + 1

if resultado == 0:
        print("\n",num," e um numero primo")
else:
        print("\n",num," nao e um numero primo")
        
print("\n\nObrigado por utilizar nosso programa!")
os.system("pause")
